### PR TITLE
added check tree in BST in c++

### LINKED ISSUE #137: Check whether a given binary tree is a Binary Search Tree
I would like to implement the solution to check if a binary tree is a Binary Search Tree in Java.
I would include test cases,complexities as well as the detailed algorithmic explanation for the same and also the common approach which leads to wrong answers.
@rudrakshi99 @nakul-19 @plazzy99 Kindly assign this to me as a part of GSSOC'21.

### PR DESCRIPTION
**Pull request by sakshi2707**
fixes #137

### CODE REVIEW

**Comment by sakshi2707** [comment]
@plazzy99 please review and merge it
also add gssoc and level labels

**Comment by rudrakshi99** [comment]
add a single algorithm in a PR.

File: Tree/readme.md [hunk 1]
```diff
@@ -47,9 +47,11 @@ that, this property should be satisfied at every node in the tree.
 * Sorting the elements of binary search tree and many more
 
 ## Questions :
-* Searching in BST ----> [C++](/Code/C++/searching_in_bst.cpp)
+* Searching in BST ----> [C++](/Code/C++/searching_in_bst.cpp) 
+* Check Whether given Binary Tree is BST ----> [C++](/Code/C++/check_tree_is_BST.cpp)
```
Comment (plazzy99) [inline Tree/readme.md:51]: `Check if a Binary Tree is a valid Binary Search Tree ----> [Java](/Code/Java/check_valid_BST.java)`

This is a previous merged pr take a pull and update readme

**Review by plazzy99 — CHANGES_REQUESTED** [review]

File: Tree/readme.md (hunk 1 above)
Comment (plazzy99) [reply Tree/readme.md:51]: this is to be done to resolve the conflict

**Comment by sakshi2707** [comment]
@plazzy99 done!

**Comment by rudrakshi99** [comment]
resolve merge conflicts

**Comment by sakshi2707** [comment]
@rudrakshi99 I've resolved all the confilcts. 
Please merge it ASAP otherwise more conflicts will be generated

**Pull request merged**